### ISSUE TITLE
URL.toString() does not produce parseable (by java.URI) URL 

### ISSUE THREAD

**Issue by dietervdw-spotify** [smola/galimatias#76]
Example of problematic url:

> http://s3.eu-central-1.amazonaws.com/dienettenumdieecke.de/Episoden/%23001%20Die%20Truman%20Show.mp3?ptm_source=feed&ptm_context=&ptm_request=38d83f1a6887&ptm_file=#001 Die Truman Show.mp3

I tried using your library to clean up this URL before feeding it to URI in order to prevent it crashing the parsing in URI, but unfortunately that doesn't work. It still produces a URL that is not parseable by the URI class.

In good news, calling `toJavaURI()` and then doing `toString()`seems to work!

(I know the conversion URI -> String -> URI sounds stupid, but for reasons I need an URI to feed in a library.)

**Comment by deansg** [comment]
Agree, another example is any url containing a pipe character ("|") in it's path string, which isn't properly escaped.
